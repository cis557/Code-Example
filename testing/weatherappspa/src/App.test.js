import React from 'react';
// import testing library functions
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import {within} from '@testing-library/dom'
import WeatherApp from './App';

// const renderer = require('react-test-renderer');

test('renders Boston link', () => {
  const { getByText } = render(<WeatherApp />);
  const linkElement = getByText(/Boston/);
  expect(linkElement).toBeInTheDocument();
});

// snapshot testing

test('Weather app matches snapshot', () => {
  const component = renderer.create(<WeatherApp />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// test type event
test('When user enters a city it is displayed', async () => {
  render(<WeatherApp />);
  // create a reference to the textbox
  const element = screen.getByRole('textbox')

  // type some text (douala) into the textbox
  await userEvent.type(element,  'douala');
  // fire a click on the Ok button
  // await user.click(screen.getByRole('button', {name: /OK/i}))
  // assertion: verify that the text is in the textbox
  expect(element).toHaveValue('douala')
});

// test click event

test('Textbox empty after clicking on a link', async () => {
  // render the component
  render(<WeatherApp />);
  // create a reference to the textbox
  const element = screen.getByRole('textbox')

  // type some text (douala) into the textbox
  await userEvent.type(element,  'douala');

  // assertion: verify that the text is in the textbox
  expect(element).toHaveValue('douala')
  // fire a click on the a link (city) button
  await userEvent.click(screen.getByText('Mumbai'));

  // assertion: verify that the textbox is empty
  expect(element).toHaveValue('')
  
});

