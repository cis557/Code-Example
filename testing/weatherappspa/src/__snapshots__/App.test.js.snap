// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather app matches snapshot 1`] = `
<div>
  <header>
    <input
      id="inpt"
      onChange={[Function]}
      type="text"
    />
    <button
      id="btn"
      onClick={[Function]}
      type="button"
    >
      OK
    </button>
  </header>
  <section>
    <h1>
      Cities
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="/Philadelphia"
            onClick={[Function]}
          >
            Philadelphia
          </a>
        </li>
        <li>
          <a
            href="/Boston"
            onClick={[Function]}
          >
            Boston
          </a>
        </li>
        <li>
          <a
            href="/Mumbai"
            onClick={[Function]}
          >
            Mumbai
          </a>
        </li>
        <li>
          <a
            href="/Dakar"
            onClick={[Function]}
          >
            Dakar
          </a>
        </li>
        <li>
          <a
            href="/Shanghai"
            onClick={[Function]}
          >
            Shanghai
          </a>
        </li>
        <li>
          <a
            href="/London"
            onClick={[Function]}
          >
            London
          </a>
        </li>
        <li>
          <a
            href="/Montreal"
            onClick={[Function]}
          >
            Montreal
          </a>
        </li>
      </ul>
    </nav>
    <article />
  </section>
</div>
`;
